rfcs/proposed/numa_support/create-numa-arenas.md

```diff
@@ -0,0 +1,110 @@
+    std::vector<tbb::task_arena> create_numa_task_arenas(
+        task_arena::constraints other_constraints = {},
+        unsigned reserved_slots = 0
+    };
+}
+```
+
+It optionally takes a `constraints` argument to change default arena settings such as maximal concurrency
+(the upper limit on the number of threads), core type etc.; the `numa_id` value in `other_constraints`
+is ignored. The second optional argument allows to override the number of reserved slots, which by default
+is 0 (unlike the `task_arena` construction default of 1) for the reasons described in the introduction.
+
+The function returns a `std::vector` of created arenas. The arenas should not be initialized,
+in order to allow changing certain arena settings before the use.
+
+### Possible implementation
+
+```c++
+std::vector<tbb::task_arena> create_numa_task_arenas(
+    tbb::task_arena::constraints other_constraints,
+    unsigned reserved_slots)
+{
+    std::vector<tbb::numa_node_id> numa_nodes = tbb::info::numa_nodes();
+    std::vector<tbb::task_arena> arenas;
+    arenas.reserve(numa_nodes.size());
+    for (tbb::numa_node_id nid : numa_nodes) {
+        other_constraints.numa_id = nid;
+        arenas.emplace_back(other_constraints, reserved_slots);
+    }
+    return arenas;
+}
+```
+
+## Considered alternatives
+
+The earlier proposal [PR #1559](https://github.com/uxlfoundation/oneTBB/pull/1559) suggested to add
+a new class derived from `task_arena` which would also provide a method to wait for work completion.
+It aims to simplify the whole usage example as much as possible. That proposal also includes
+a factory function to create a set of such arenas constrained by NUMA domains.
+
+While suggesting very concise and "bullet-proof" API for the problem at question, that approach has
+its downsides:
+- Adding a special "flavour" of `task_arena` potentially increases the library learning curve and
+  might create confusion about which class to use in which conditions, and how these interoperate.
+- It seems very specialized, capable to only address specific and quite narrow set of use cases.
+
+Our proposal, instead, aims at a single usability aspect with incremental improvements/extensions
+to the existing oneTBB classes and usage patterns, leaving other aspects to complementary proposals.
+Specifically, the [Waiting in a task arena](../task_arena_waiting/readme.md) RFC improves the joint
+use of a task arena and a task group to submit and wait for work. Combined, these extensions will
+provide only a slightly more verbose solution for the NUMA use case, while being more flexible
+and having greater potential for other useful extensions and applications.
+
+## Open questions
+- Instead of a free-standing function in namespace `tbb`, should we consider
+  a static member function in class `task_arena`?
+- The proposal does not consider arena priority, simply keeping the default `priority::normal`.
+  Should a possibility to change priorities be considered?
```

Comment:
We also have `leave_policy` setting as a preview feature now.

Reply:
Thanks for reminding me of it :)
To me, this is a reason not to extend the list of parameters beyond what is seen practically useful, at least for now.

Reply:
Expanded the question to mention the leave policy as well.